pkg/scheduler/scheduler.go

```diff
@@ -253,6 +252,26 @@ func New(client clientset.Interface,
 	for _, opt := range opts {
 		opt(&options)
 	}
+
+	schedulerCache := internalcache.New(30*time.Second, stopCh)
+	volumeBinder := volumebinder.NewVolumeBinder(client, nodeInformer, pvcInformer, pvInformer, storageClassInformer,
+		time.Duration(options.bindTimeoutSeconds)*time.Second)
+
+	if options.frameworkRegistry == nil {
```

Comment:
If external users register something into `frameworkRegistry`, i.e. it's not nil, IMO it's still possible for the internal scheduler to register the `frameworkplugins.RegistryArgs` below? Are you concerned with some side effects? If so, we can warn users.

Reply:
I mean we can still give a bare `frameworkRegistry` to end users to register custom things at `cmd/.../app.go`, and then inject the necessary fields here.

Reply:
I think I know how to do this. We can add an option to scheduler.New to pass a "outOfTreeRegistry", which collects out-of-tree plugins that we append to the default registry, I will send a patch to show you how it looks, thanks for brainstorming.